lib/test/unit/ui/tap/ext/base_testrunner.rb

```diff
@@ -0,0 +1,62 @@
+require 'test/unit/ui/tap/base_testrunner'
+
+module Test
+  module Unit
+    module UI
+      module Tap
+        class Tap::BaseTestRunner < Test::Unit::UI::TestRunner
+
+          def tapout_pass_with_source_location(test)
+            doc = tapout_pass_without_source_location(test)
+            if(doc && doc['type'] == 'test')
+              file, _ = test.class.instance_method(clean_label(test.name).to_sym).source_location
+              doc['file'] = file.sub(Dir.pwd+'/', '')
+            end
+            doc
+          end
+
+          def tapout_todo_with_source_location(fault)
+            doc = tapout_todo(fault)
```

Comment:
Thanks for catching this. It was running in an infinite loop. 

I've fixed this in https://github.com/renderedtext/test-unit-runner-tap/commit/9454d2821bb26e0f9880188a1a6eeda5616d96a4 and have added a test.